data-prepper-plugins/http-source/src/main/java/org/opensearch/dataprepper/plugins/source/loghttp/LogHTTPService.java

```diff
@@ -101,7 +101,7 @@ private HttpResponse processRequest(final AggregatedHttpRequest aggregatedHttpRe
         List<List<String>> jsonList;
 
         try {
-            jsonList = (maxRequestLength == null) ? jsonCodec.parse(content) : jsonCodec.parse(content, maxRequestLength - SERIALIZATION_OVERHEAD);
+            jsonList = (maxRequestLength == null) ? jsonCodec.parse(content) : jsonCodec.parse(content, buffer.getOptimalRequestSize().get() - SERIALIZATION_OVERHEAD);
```

Comment:
There are two small problems here that you can fix.

```
buffer.getOptimalRequestSize().get() 
```

1. `buffer.getOptimalRequestSize()` may return `null`, leading to NPE on `.get()`
2. `buffer.getOptimalRequestSize()` may return empty, leading to a `NoSuchElementException` on `.get()`.

Reply:
I don't think you need an NPE check actually. Just check that the optional is present.